cinema/views.py

```diff
@@ -0,0 +1,41 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from rest_framework import status
+
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+def movie_create_or_update(data, instance=None):
+    if instance:
+        serializer = MovieSerializer(instance, data)
+    else:
+        serializer = MovieSerializer(data=data)
+    serializer.is_valid(raise_exception=True)
+    serializer.save()
+    return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+@api_view(["GET", "POST"])
+def movie_list(request):
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    return movie_create_or_update(request.data)
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, pk):
+    movie = get_object_or_404(Movie, pk=pk)
+    if request.method == "GET":
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    if request.method == "PUT":
+        return movie_create_or_update(request.data, movie)
```

Comment:
It would be better to separate `Movie` creation and and update between `POST` and `PUT` methods correspondingly and to not delegate both actions to `PUT`.  Follow this approach in the future